src/sentry/issues/escalating.py

```diff
@@ -312,7 +312,7 @@ def manage_snooze_states(
         group.substatus = GroupSubStatus.ESCALATING
         group.status = GroupStatus.UNRESOLVED
```

Comment:
I think we might need to follow the same pattern to conditionally update the Group status and substatus, instead of using `Group.save()` here. 

https://github.com/getsentry/sentry/blob/4833f87760fca3ee42507c2accba9ed424c516ca/src/sentry/event_manager.py#L1773-L1793

Since this function gets called in post_process_group per event that groups to the issue, we might run into races where two events belonging to the same group will trigger the same side-effects when we really only want it to be triggered once.